src/components/scene/background.js

```diff
@@ -15,6 +16,30 @@ module.exports.Component = register('background', {
       return;
     }
     object3D.background = new THREE.Color(data.color);
+
+    if (data.generateEnvironment) {
+      const scene = this.el.sceneEl.object3D;
+      const cubeRenderTarget = new THREE.WebGLCubeRenderTarget(128, { format: THREE.RGBFormat, generateMipmaps: true, minFilter: THREE.LinearMipmapLinearFilter });
+      const cubeCamera = new THREE.CubeCamera(1, 100000, cubeRenderTarget);
+      this.cubeCamera = cubeCamera;
+      this.needsEnvironmentUpdate = true;
+      scene.environment = cubeRenderTarget.texture;
+      this.el.sceneEl.addEventListener('loaded', function () {
+        this.needsEnvironmentUpdate = true;
+      });
+    }
+  },
+
+  tick () {
+    if (!this.needsEnvironmentUpdate) return;
+    const scene = this.el.object3D;
```

Comment:
replace with var (A-Frame is ES5)

Reply:
Haha oh wow blast from the past!

I should also change `tick () {}` too as well!